jx_sqlite/aggs_table.py

```diff
@@ -525,6 +525,22 @@ def _groupby_op(self, query, frum):
             nested_path: "__" + unichr(ord('a') + i) + "__"
             for i, (nested_path, sub_table) in enumerate(self.sf.tables.items())
             }
+        frum_path = split_field(frum)
+        base_table = join_field(frum_path[0:1])
+        path = join_field(frum_path[1:])
+        tables = []
+        for n, a in nest_to_alias.items():
+            if startswith_field(path, n):
+                tables.append({"nest": n, "alias": a})
+        tables = jx.sort(tables, {"value": {"length": "nest"}})
+
+        from_sql = join_field([base_table] + split_field(tables[0].nest)) + " " + tables[0].alias
+        previous = tables[0]
+        for t in tables[1::]:
+            from_sql += (
+                "\nLEFT JOIN\n" + quote_table(concat_field(base_table, t.nest)) + " " + t.alias +
+                " ON " + t.alias + "." + PARENT + " = " + previous.alias + "." + UID
+            )
```

Comment:
`_edges_op()` and `_groupby_op()` are very similar, and they can each "learn" from each other; maybe even identify bugs between them.

I do not like code-duplication, but I realize code-duplication can be used to for both code clarity and to reduce code dependencies.  This is one of those cases that sits between; we want the code to be independent because it is different in subtle ways, but it is also very similar.  In this case, we want to perform line-by-line comparision between these two methods.

Please split `aggs_table.py` into two modules: `edges_table.py` and `groupby_table.py`.  Have the class of one inherit from the other, and ensure they are laid out almost-identical so you can perform a file comparison between the two.